kernel/kernel-api/src/test/scala/io/delta/kernel/internal/TableConfigSuite.scala

```diff
@@ -0,0 +1,163 @@
+import java.util.{Collections, Optional}
+import scala.collection.JavaConverters._
+
+class TableConfigSuite extends AnyFunSuite with MockEngineUtils {
+
+  test("Parse Map[String, String] type table config") {
+    val expMap = Map("key1" -> "string_value", "key2Int" -> "2", "key3ComplexStr" -> "\"hello\"")
+    val engine = mockEngine(jsonHandler = new KeyValueJsonHandler(expMap))
+    assert(
+      TableConfig.parseJSONKeyValueMap(
+        engine,
+        """{"key1": "string_value", "key2Int": "2", "key3ComplexStr": "\"hello\""}""") ===
+        expMap.asJava)
+
+    val engine1 = mockEngine(jsonHandler = new ReturnNullRowsJsonHandler(0))
+    val e1 = intercept[IllegalStateException] {
+      TableConfig.parseJSONKeyValueMap(engine1, """{"key": "value"}""")
+    }
+    assert(e1.getMessage.contains("""Unable to parse {"key": "value"}"""))
+
+    val engine2 = mockEngine(jsonHandler = new ReturnNullRowsJsonHandler(2))
+    val e2 = intercept[IllegalArgumentException] {
+      TableConfig.parseJSONKeyValueMap(engine2, """{"key": "value"}""")
+    }
+    assert(e2.getMessage.contains("Iterator contains more than one element"))
+
+    val errMsg = "Close called failed"
+    val engine3 = mockEngine(jsonHandler = new ReturnCloseFailIteratorJsonHandler(errMsg))
+    val e3 = intercept[KernelException] {
+      TableConfig.parseJSONKeyValueMap(engine3, """{"key": "value"}""")
+    }
+    assert(e3.getMessage.contains(s"java.io.IOException: $errMsg"))
+
+    val engine4 = mockEngine(jsonHandler = new ReturnNullRowsJsonHandler(1))
+    val e4 = intercept[IllegalArgumentException] {
+      TableConfig.parseJSONKeyValueMap(engine4, """{"key": "value"}""")
+    }
+    assert(e4.getMessage === null)
+  }
+}
+
+/**
+ * Mock JsonHandler which returns a ColumnarBatch with a single row containing a Map[String, String]
+ * column.
+ */
+class KeyValueJsonHandler(map: Map[String, String]) extends BaseMockJsonHandler {
+  val expSchema: StructType =
+    new StructType().add("config", new MapType(StringType.STRING, StringType.STRING, true))
+  override def parseJson(
+    jsonStringVector: ColumnVector,
+    outputSchema: StructType,
+    selectionVector: Optional[ColumnVector]): ColumnarBatch = {
+    assert(outputSchema == expSchema)
+    new ColumnarBatch {
+      override def getSchema: StructType = outputSchema
+
+      override def getColumnVector(ordinal: Int): ColumnVector = {
+        ordinal match {
+          case 0 => mapTypeVector(map)
+        }
+      }
+
+      override def getSize: Int = 1
+    }
+  }
+
+  def mapTypeVector(map: Map[String, String]): ColumnVector = {
+    new ColumnVector {
+      override def getDataType: DataType = new MapType(StringType.STRING, StringType.STRING, true)
+
+      override def getSize: Int = 1
+
+      override def close(): Unit = {}
+
+      override def isNullAt(rowId: Int): Boolean = rowId >= 1
+
+      override def getMap(rowId: Int): MapValue = VectorUtils.stringStringMapValue(map.asJava)
+    }
+  }
```

Comment:
You can actually just use to the `DefaultGenericVector` with size = 1

Reply:
We could not use it because it's defined in `Kernel-Default`?

Reply:
Moved it to `VectorTestUtils`.